### PR TITLE
added charset changes from go.enmime

### PR DESCRIPTION
**Pull request by csucu**
Merged in following commits:
https://github.com/redsift/go.enmime/commit/6b76a8110a8a64a706f53edd88370e452263cc0e
https://github.com/redsift/go.enmime/commit/35192b153ee4d06471fb2a6f940c22371a1ed92a
https://github.com/redsift/go.enmime/commit/5a74234946317a0487174fdaab85576a6afdee53
https://github.com/redsift/go.enmime/commit/fa72f277d84a5c793a6c3ca579a1e0ff5533c00c
https://github.com/redsift/go.enmime/commit/8754fade0af9a750fb4852887b316eca5716c12b
https://github.com/redsift/go.enmime/commit/57c878e0d07ab15242abe6fc604e78105ecb0db4

### CODE REVIEW

File: part.go [hunk 1]
```diff
@@ -231,7 +231,8 @@ func (p *Part) convertFromStatedCharset(r io.Reader) io.Reader {
 	// Try to parse charset again here to see if we can salvage some badly formed
 	// ones like charset="charset=utf-8".
 	charsetp := strings.Split(p.Charset, "=")
-	if strings.ToLower(charsetp[0]) == "charset" && len(charsetp) > 1 {
+	if (strings.ToLower(charsetp[0]) == "charset" && len(charsetp) > 1) ||
```
Comment (amnonbc) [inline part.go]: maybe `strings.EqualFold()` is clearer?

**Review by amnonbc — APPROVED** [review]

**Review by dmotylev — APPROVED** [review]
LGTM

File: part.go (hunk 1 above)
Comment (dmotylev) [reply part.go]: It is even faster; no allocations

Comment (csucu) [reply part.go]: done

**Pull request merged**